app/rdf/query-hierarchies.ts

```diff
@@ -77,13 +88,13 @@ export const queryHierarchy = async (
   sparqlClient: ParsingClient,
   sparqlClientStream: StreamClient
 ): Promise<HierarchyValue[] | null> => {
-  const hierarchy = findHierarchyForDimension(
+  const hierarchies = findHierarchiesForDimension(
     rdimension.cube,
     rdimension.data.iri
   );
 
   // @ts-ignore
-  if (!isGraphPointer(hierarchy)) {
+  if (!isGraphPointer(hierarchies[0])) {
     return null;
   }
```

Comment:
What if `hierarchies[n]` for (n > 0) is a GraphPointer? Should this logic be looped over all fetched hierarchies?

Reply:
I changed the conditions to check the length of the array as the goal of the function was to check if there was any hierarchy.